src/Analyzer/Resolve/IdentifierLookup.h

```diff
@@ -190,6 +204,15 @@ struct IdentifierResolveSettings
 
     /// Allow to resolve subquery during identifier resolution
     bool allow_to_resolve_subquery_during_identifier_resolution = true;
+
+    IdentifierResolveScope * scope_to_resolve_alias_expression = nullptr;
+
+    IdentifierResolveContext & resolveAliasesAt(IdentifierResolveScope * scope_to_resolve_alias_expression_)
```

Comment:
I see only one usage of this function; maybe remove it

Reply:
I'd keep it at least for now :)